### PR TITLE
Check `cfg` during CI and fix feature typos

### LINKED ISSUE #12091: Check `cfg` to catch feature typos
Nightly Rust has added support for `-Zcheck-cfg`, which will show a warning if there is a `#[cfg(...)` for a nonexistant feature / cfg.

While not possible to add this to CI because it is currently nightly-only, it can be run locally to catch all current typos.

Please read [this comment](https://github.com/rust-lang/rfcs/pull/3013#issuecomment-1936648479) on how to use `-Zcheck-cfg`. Once you follow the setup, run:

```shell
cargo +nightly check -Zcheck-cfg --workspace
```

I found a few warnings related to missing features, but I currently don't have the time to work on it today.

### PR DESCRIPTION
**Pull request by eerii**
# Objective

- Add the new `-Zcheck-cfg` checks to catch more warnings
- Fixes #12091

## Solution

- Create a new `cfg-check` to the CI that runs `cargo check -Zcheck-cfg --workspace` using cargo nightly (and fails if there are warnings)
- Fix all warnings generated by the new check

---

## Changelog

- Remove all redundant imports
- Fix cfg wasm32 targets
- Add 3 dead code exceptions (should StandardColor be unused?)
- Convert ios_simulator to a feature (I'm not sure if this is the right way to do it, but the check complained before)

## Migration Guide

No breaking changes

### CODE REVIEW

File: crates/bevy_color/src/lib.rs [hunk 1]
```diff
@@ -98,6 +98,7 @@ pub use xyza::*;
 use bevy_render::color::LegacyColor;
 
 /// Describes the traits that a color should implement for consistency.
+#[allow(dead_code)]
```
Comment (alice-i-cecile) [inline crates/bevy_color/src/lib.rs:101]:
```suggestion
#[allow(dead_code)] // This is an internal marker trait used to ensure that our color types impl the required traits
```

File: crates/bevy_reflect/src/serde/de.rs [hunk 2]
```diff
@@ -23,6 +23,7 @@ pub trait DeserializeValue {
 }
 
 trait StructLikeInfo {
+    #[allow(dead_code)]
```
Comment (alice-i-cecile) [inline crates/bevy_reflect/src/serde/de.rs:26]: @MrGVSV @soqb what do you think should be done about these dead-code warnings?

**Comment by alice-i-cecile** [comment]
Wow, this is an incredibly useful check. There are some nasty subtle bugs in there.

File: crates/bevy_reflect/src/serde/de.rs (hunk 2 above)
Comment (MrGVSV) [reply crates/bevy_reflect/src/serde/de.rs:26]: If it's not being used, we can just get rid of it. These traits are private to this one module

**Review by james7132 — APPROVED** [review]
This is great. I'm a bit concerned about false positives being a bit too aggressive, but this is really useful.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Once we remove the dead code from bevy_reflect this LGTM.

File: crates/bevy_reflect/src/serde/de.rs (hunk 2 above)
Comment (soqb) [reply crates/bevy_reflect/src/serde/de.rs:26]: go ahead and remove them. i don't know whether it was the `TypePath` changes or the serialization format changes which made these methods unnecessary but we don't need the path here anymore.

Comment (eerii) [reply crates/bevy_reflect/src/serde/de.rs]: Removed them in the last commit!

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**